packages/dds/tree/src/feature-libraries/mapTreeCursor.ts

```diff
@@ -24,7 +24,18 @@ export function singleMapTreeCursor(root: MapTree): ITreeCursorSynchronous {
 const adapter: CursorAdapter<MapTree> = {
     value: (node) => node.value,
     type: (node) => node.type,
-    keysFromNode: (node) => [...node.fields.keys()], // TODO: don't convert this to array here.
+    keysFromNode: (node) => {
```

Comment:
like jsonableTree, all current users of MapTree should avoid storing empty field arrays, so we shouldn't need to do anything special here. I checked and that isn't documented on MapTree: you should be able to ad docs for that instead of handling it here.

Also, if you were going to keep this code, there is no reason for it to call isArray: the value type in MapTree is MapTree[]: you just found a slower way to cast away the fact that get can return undefined. A better way to do that would be to iterate over the map instead of its keys and calling get for each one.